animation_nodes/nodes/matrix/distribute_matrices.pyx

```diff
@@ -139,14 +141,24 @@ class DistributeMatricesNode(bpy.types.Node, AnimationNode):
             if self.splineDistributionMethod != "VERTICES":
                 self.newInput("Float", "Start", "start", value = 0.0).setRange(0.0, 1.0)
                 self.newInput("Float", "End", "end", value = 1.0).setRange(0.0, 1.0)
+        elif self.mode == "HEXAGON":
+            self.newInput("Integer", "X Divisions", "xDiv", value = 6, minValue = 0)
+            self.newInput("Integer", "Y Divisions", "yDiv", value = 6, minValue = 0)
+            if self.distanceMode == "STEP":
+                self.newInput("Float", "X Distance", "size1", value = 1)
+                self.newInput("Float", "Y Distance", "size2", value = 1)
+            elif self.distanceMode == "SIZE":
+                self.newInput("Float", "Width", "size1", value = 5)
+                self.newInput("Float", "Length", "size2", value = 5)
+            self.newInput("Float", "Z Scale", "size3", value = 1)
```

Comment:
I see no reason to keep this here. The user should scale the matrices manually after distribution or supply a properly scaled object. So I think this should be omitted from this node.

Reply:
removed

Reply:
Perhaps you misunderstood my intentions here. I don't think we should scale the matrices in the node. Just like the grid node, where the matrices are always unity scaled regardless of the distribution. It follows that the Z scale input should be removed from the node.

Reply:
Ok i'll remove scale from matrix. So, what about ``Step`` and ``Size`` option should i keep or remove it?

Reply:
Keep it.

Reply:
> Keep it.

Can i reuse ``distanceMode`` from grid ?

Reply:
It is better if you don't.

Reply:
@OmarEmaraDev I have changed the code.